2-mandatory/2-financial-times.js

```diff
@@ -14,68 +23,102 @@ function potentialHeadlines(allArticleTitles) {
     (you can assume words will always be seperated by a space)
 */
 function titleWithFewestWords(allArticleTitles) {
-    // TODO
+  let min = Infinity;
+  let tempIndex;
+  for (let i = 0; i < allArticleTitles.length; i++) {
+    if (allArticleTitles[i].length < min) {
+      min = allArticleTitles[i].length;
+      tempIndex = i;
+    }
+  }
+  return allArticleTitles[tempIndex];
 }
 
 /*
     The editor of the FT has realised that headlines which have numbers in them get more clicks!
     Implement the function below to return a new array containing all the headlines which contain a number.
     (Hint: remember that you can also loop through the characters of a string if you need to)
 */
+
 function headlinesWithNumbers(allArticleTitles) {
-    // TODO
+  let returnedArray1 = [];
+
+  for (let element of allArticleTitles) {
+    //to loop through the AllArticleTitles
+    let [...tester] = element; // to convert element into an array called tester, then test each char
+
+    for (let numChecker of tester) {
+      // to loop through each element
+      if (numChecker >= 0 && numChecker !== " ") {
```

Comment:
It's a weird JavaScript thing that a string is less than 0 and it's not very obvious to the person reading the code. It would be better to use some regex here to check if something is a number, like `char.match(/[0123456789]/g)`. Melese had a good answer for this one, you can see it here https://github.com/CodeYourFuture/JavaScript-Core-1-Coursework-Week3/pull/239/files#r1132433282.

Reply:
I was trying to check if this variable is a number. I'll try regex instead. Thanks.